persistence.js

```diff
@@ -29,19 +32,26 @@ function RedisPersistence (opts) {
 
   opts = opts || {}
   this.maxSessionDelivery = opts.maxSessionDelivery || 1000
+  this.packetTTL = opts.packetTTL || function () { return 30 * 24 * 60 * 60 * 1000 }
+
   this._db = new Redis(opts)
 
   this.messageIdCache = HLRU(100000)
 
   var that = this
-  this._decodeAndAugment = function decodeAndAugment (chunk, enc, cb) {
-    that._db.getBuffer(chunk, function decodeMessage (err, result) {
-      var decoded
-      if (result) {
-        decoded = msgpack.decode(result)
-      }
-      cb(err, decoded)
-    })
+  this._decodeAndAugment = function (listKey) {
+    return function decodeAndAugment (key, enc, cb) {
+      that._db.getBuffer(key, function decodeMessage (err, result) {
+        var decoded
+        if (result) {
+          decoded = msgpack.decode(result)
+        }
+        if (err || !decoded) {
+          that._db.lrem(listKey, 0, key)
+        }
+        cb(err, decoded)
+      })
+    }
```

Comment:
this was done to avoid having to allocate a new closure for each method. It was a bit silly anyway, so why don't you move this to a _buildAugment method attached to the prototype?